jkube-kit/generator/api/src/test/java/org/eclipse/jkube/generator/api/support/BaseGeneratorTest.java

```diff
@@ -317,83 +302,63 @@ public void shouldNotAddDefaultImageInCaseOfSimpleDockerfile() throws IOExceptio
     File projectBaseDir = folder.newFolder("test-project-dir");
     File dockerFile = new File(projectBaseDir, "Dockerfile");
     boolean isTestDockerfileCreated = dockerFile.createNewFile();
-    new Expectations() {
-      {
-        ctx.getProject();
-        result = project;
-        project.getBaseDirectory();
-        result = projectBaseDir;
-      }
-    };
-
+    when(ctx.getProject()).thenReturn(project);
+    when(project.getBaseDirectory()).thenReturn(projectBaseDir);
     // When
     BaseGenerator generator = createGenerator(null);
 
     // Then
-    assertTrue(isTestDockerfileCreated);
-    assertFalse(generator.shouldAddGeneratedImageConfiguration(Collections.emptyList()));
+    assertThat(isTestDockerfileCreated).isTrue();
+    assertThat(generator.shouldAddGeneratedImageConfiguration(Collections.emptyList())).isFalse();
   }
 
   @Test
   public void shouldAddGeneratedImageConfiguration_whenAddEnabledViaConfig_shouldReturnTrue() {
     // Given
-    new Expectations() {{
-      ctx.getProject();
-      result = project;
-    }};
+    when(ctx.getProject()).thenReturn(project);
     properties.put("jkube.generator.test-generator.add", "true");
     BaseGenerator generator = createGenerator(null);
 
     // When
     boolean result = generator.shouldAddGeneratedImageConfiguration(createNewImageConfigurationList());
 
     // Then
-    assertTrue(result);
+    assertThat(result).isTrue();
   }
 
 
   @Test
   public void shouldAddGeneratedImageConfiguration_whenAddEnabledViaProperty_shouldReturnTrue() {
     // Given
-    new Expectations() {{
-      ctx.getProject();
-      result = project;
-    }};
+    when(ctx.getProject()).thenReturn(project);
     properties.put("jkube.generator.add", "true");
     BaseGenerator generator = createGenerator(null);
 
     // When
     boolean result = generator.shouldAddGeneratedImageConfiguration(createNewImageConfigurationList());
 
     // Then
-    assertTrue(result);
+    assertThat(result).isTrue();
   }
 
   @Test
   public void addLatestTagIfSnapshot() {
-    new Expectations() {
-      {
-        ctx.getProject();
-        result = project;
-        project.getVersion();
-        result = "1.2-SNAPSHOT";
-      }
-    };
+    when(ctx.getProject()).thenReturn(project);
+    when(project.getVersion()).thenReturn("1.2-SNAPSHOT");
     BuildConfiguration.BuildConfigurationBuilder builder = BuildConfiguration.builder();
     BaseGenerator generator = createGenerator(null);
     generator.addLatestTagIfSnapshot(builder);
     BuildConfiguration config = builder.build();
     List<String> tags = config.getTags();
-    assertEquals(1, tags.size());
-    assertTrue(tags.get(0).endsWith("latest"));
+    assertThat(tags)
```

Comment:
I'm going to approve this PR because we've iterated much over it, but this was not the suggested change.
https://github.com/eclipse/jkube/pull/1657#discussion_r936604328
this is completely wrong, you should assert the opposite.
Anyway, you should be able to combine this assertion and the next

```java
assertThat(tags).singleElement().endsWith("latest");
```
The calls to satisfies should be used as the last resort, and in this case, there is a much simpler approach.